ledger-core/virtual/object/object_migration_test.go

```diff
@@ -109,21 +115,29 @@ func TestSMObject_MigrationCreateStateReport_IfStateIsEmptyAndNoCounters(t *test
 		UnpublishAllMock.Return().
 		ShareMock.Set(
 		func(data interface{}, flags smachine.ShareDataFlags) (s1 smachine.SharedDataLink) {
-			require.IsType(t, &payload.VStateReport{}, data)
+			switch d := data.(type) {
+			case payload.VStateReport:
+				require.IsType(t, &payload.VStateReport{}, d)
```

Comment:
oof, bad

Reply:
Explain, please

Reply:
hmm. Why do you need to check for type after type assertion? Do you not want to test type assertion?

Reply:
and if d is supposed to be &payload.VStateReport{}, then type assertion should be 
```
case *payload.VStateReport:
```

Reply:
We need to skip other types check

Reply:
I mean after type assert - you are pretty sure what type d is. What require supposed to do?

Reply:
Fixed